### ISSUE TITLE
Plugin is trying to push to Registry v1 instead of v2

### ISSUE THREAD

**Issue by Vad1mo** [bmuschko/gradle-docker-plugin#202]
Hello, I am trying to push images with gradle-docker-plugin into a private docker registry running docker registry 2.3.1 (hence v2)

on the command line I can successfully bush images but not with the plugin. 

This is the error I always get: 
```
* What went wrong:
Execution failed for task ':web:pushImage'.
> Could not push image: Get https://registry.corp.acme.com/v1/_ping: dial tcp 172.16.21.2:443: i/o timeout

* Try:
Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':web:pushImage'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:69)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:46)
	at 
```

my `build.gradle`
```
apply from: '../docker-common.gradle'
import com.bmuschko.gradle.docker.tasks.image.*
def p21Registry = 'registry.corp.acme.com/acme'

task buildImage(type: DockerBuildImage, dependsOn: [copyVersion,processResources]) {
    inputDir = project.projectDir
    tag = "$p21Registry/$rootProject.name"
}

task tagImage(type: DockerTagImage, dependsOn: buildImage) {
    imageId buildImage.getTag()
    repository = "$p21Registry/$rootProject.name"
    tag = version
}

task pushImage(type: DockerPushImage, dependsOn: tagImage) {
    imageName = "$p21Registry/$rootProject.name:$version"
}
```

my `docker-common.gradle` looks like this

```
docker {
    if (System.env.DOCKER_HOST) {
        url = "$System.env.DOCKER_HOST".replace("tcp", "https")
        certPath = new File(System.env.DOCKER_CERT_PATH)
    } else {
        url = 'unix:///var/run/docker.sock'
    }

    registryCredentials {
        url = 'https://registry.corp.acme.com/v2'
        username = project.hasProperty("docker_username") ? docker_username : System.env.DOCKER_USERNAME
        password = project.hasProperty("docker_password") ? docker_password: System.env.DOCKER_PASSWORD
        email = project.hasProperty("docker_email") ? docker_email: System.env.DOCKER_EMAIL
    }
}
```

I am using the plugin version 2.6.8 btw.

**Comment by cdancy** [comment]
@Vad1mo you have to explicitly define the [api version](https://github.com/bmuschko/gradle-docker-plugin#extension-properties) you are using via the extension (i.e. api = "v1.19") to talk with a v2 registry. This will be fixed in a future release but for now it's what is needed. Take a look at #182 to understand the history and how to move forward. Let me know how things work out.

**Comment by Vad1mo** [comment]
Thank you @cdancy I have set the version to 1.22. Didn't notice this option previously

But now I get this error here:
```
[pool-2-thread-1] ERROR com.github.dockerjava.core.async.ResultCallbackTemplate - Error during callback
com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "aux" (class com.github.dockerjava.api.model.PushResponseItem), not marked as ignorable (9 known properties: , "stream", "error", "from", "progress", "time", "status", "errorDetail", "id", "progressDetail"])
 at [Source: N/A; line: -1, column: -1] (through reference chain: com.github.dockerjava.api.model.PushResponseItem["aux"])
	at com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException.from(UnrecognizedPropertyException.java:79)
	at com.fasterxml.jackson.databind.DeserializationContext.reportUnknownProperty(DeserializationContext.java:579)
```

Searching on the internet I found this issue:
https://github.com/docker-java/docker-java/issues/437


What docker lib version is this plugin using? I can't find the dependency

**Comment by cdancy** [comment]
@Vad1mo we are using 2.2.3. They currently have an in-development v3 coming out soon which will solve a number of issues. However, we won't rebase off of that until they've gone GA.

**Comment by Vad1mo** [comment]
Thank you. Browsing the docker-java issues https://github.com/docker-java/docker-java/issues/552 and https://github.com/docker-java/docker-java/issues/506 it will take some weeks for the next release. 

Do you have an idea how can we workaround this issue until a new version of the plugin is released?

**Comment by cdancy** [comment]
What version of docker are you using? I suspect it's something fairly recent. We are using docker 1.9 here without any issues.

**Comment by Vad1mo** [comment]
I tried it with 1.11 and 1.10 but we can't go down to 1.9 as we are using some 1.10 features. 

I tried to set the apiVersion to `1.19` and `1.22` and `1.23` In all ceses it faild with this exception:

```
ERROR com.github.dockerjava.core.async.ResultCallbackTemplate - Error during callback
com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "aux" (class 
com.github.dockerjava.api.model.PushResponseItem), not marked as ignorable (9 known properties: , 
"stream", "error", "from", "progress", "time", "status", "errorDetail", "id", "progressDetail"])
```

**Comment by cdancy** [comment]
This is a known issue with docker-java and not being able to handle non-previously defined fields. What version of this plugin are you using? I believe this was fixed but may have been done only on their 3.0.0 branch.

**Comment by Vad1mo** [comment]
I am using 2.6.8 of gradle-docker-plugin. This fix was scheduled for for docker java 3.0.

For now I use the docker command to push the build images.
